model_access_restriction/models/ir_model_access_restriction.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2025 Alberto Martínez <alberto.martinez@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, exceptions, fields, models
+
+
+class IrModelAccessRestriction(models.Model):
+    _name = "ir.model.access.restriction"
+    _description = "Model Access Restrictions. Not having one will disable access."
+
+    _order = "model_id,name,id"
+
+    name = fields.Char(index=True)
+    active = fields.Boolean(
+        default=True,
+    )
+    model_id = fields.Many2one(
+        string="Model",
+        comodel_name="ir.model",
+        index=True,
+        required=True,
+        ondelete="cascade",
+        domain=lambda self: [
+            (
+                "id",
+                "!=",
+                self.env.ref(
+                    "model_access_restriction.model_ir_model_access_restriction"
+                ).id,
+            )
+        ],
+    )
+    groups = fields.Many2many(
+        string="Allowed Groups",
+        comodel_name="res.groups",
+        relation="model_access_restriction_group_rel",
+        column1="model_access_restriction_id",
+        column2="group_id",
+        ondelete="restrict",
+    )
+    # perm_read = fields.Boolean(string='Apply for Read', default=True)
```

Comment:
The commented code should not be uploaded. If it is uploaded, the reason must be indicated.